db/sql/0_msar.sql

```diff
@@ -308,9 +348,64 @@ END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
+-- Drop columns from table -------------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION
+__msar.drop_columns(table_name text, columns variadic text[]) RETURNS text AS $$/*
+Drop the given columns from the given table.
+
+Args:
+  table_name: fully-qualified, quoted table name.
+  columns: The column names to be dropped, quoted.
+*/
+DECLARE column_drops text;
+BEGIN
+  SELECT string_agg(format('DROP COLUMN %s', col), ', ')
+  FROM unnest(columns) AS col
+  INTO column_drops;
+  RETURN __msar.exec_ddl('ALTER TABLE %s %s', table_name, column_drops);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.drop_columns(table_id oid, attnums variadic integer[]) RETURNS text AS $$/*
+Drop the given columns from the given table.
+
+Args:
+  table_id: OID of the table whose columns we'll drop.
+  attnums: The attnums of the columns to drop.
+*/
+DECLARE columns text[];
+BEGIN
+  SELECT array_agg(quote_ident(attname))
+  FROM pg_attribute
+  WHERE attrelid=table_id AND ARRAY[attnum::integer] <@ attnums
+  INTO columns;
+  RETURN __msar.drop_columns(__msar.get_table_name(table_id), variadic columns);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.drop_columns(schema_ text, table_ text, columns variadic text[]) RETURNS text AS $$/*
```

Comment:
Any reason why here you use `x_` while in the above functions you use `x_name` or `x_id`? Also, in the first dispatch, you use `columns variadic text[]`, which, given the signature, does convey what identifiers these are, but not saying it in the name is inconsistent with `table_name` and `column_id`. I think the naming consistency could be improved. I like the `[qualifier]_[id type]` scheme, as in `table_name`.

Reply:
Yes, but I acknowledge I've made some compromises. Perhaps we can improve by putting our heads together. There are a few places where these names are used. 
- In the code itself: For this, more verbose might be okay (more later on that)
- In the `Argument data types` column of the output from `\df the_func_name` in `psql`.
- In the docstring of the function (which appears in the `\df+ the_func_name` output along with the code).

Long variable names are a major liability for the last two. I've (not completely arbitrarily) chosen to hard-limit line widths to 100 characters. The reason for that is that the output of descriptive functions and commands in `psql` is always _tabular_, and there isn't much point in describing functions or names at all if it makes the tabular output too wide (since it'll be quite illegible). 

Long function signatures really strain the output of `\df` and `\df+` in this respect. In fact, naming variables at all is not that common. I chose to use names, but try pretty hard to lean on types so the user could tell what they refer to. (I.e., if there are two IDs, I specify which is which; If there are two names, I give, e.g., `schema_, table_` so the next person can at a glance figure out which `TEXT` entry is which, etc.) 

You're right that my current setup is a bit inconsistent. The inconsistency crept in when creating functions that map between ID types (e.g., you don't want to use `table_` for the OID and name in the same function (and can't, in SQL)). I ended up going with adding the `_id` suffix for numeric IDs, but not `_name` for names. Again, horizontal space is generally at a premium wherever the signature is actually viewed by a user (as compared to the developer).

Reply:
@dmos62 I've tried a different variable naming setup, explained in a comment at the top of the `0_msar.sql` file. Let's see what you think of it.